Replace hard coded location protocol "https" with the one derived fro…

jeddcuison opened the pull request:
The web server balance manager link is always set to use "https" since it was hard coded. This change replaces the hard coded https with a location protocol that was derived from the web server status path. If the status path is relative, the location protocol will be the location protocol of the current URL address.

jeddcuison committed:
Replace hard coded location protocol "https" with the one derived from the web server status path

jeddcuison committed:
Test balancer manager status button using Selenium

jmahmoodcc commented on jwala-webapp/src/main/webapp/resources/js/react/group-operations/WebServerControlPanelWidget.js:
```diff
@@ -134,7 +137,23 @@ var WebServerControlPanelWidget = React.createClass({
+        parseUrlProtocol: function(url) {
+            try {
+                let protocol = url.split("/")[0].toLowerCase();
+                if (protocol.indexOf(WebServerControlPanelWidget.HTTP_PROTOCOL) > -1) {
```
Since this is testing for "http:", isn't it possible that "https" won't be returned either if the location.protocol is "http" too?

jeddcuison replied:
Good catch! I changed the code.

jmahmoodcc reviewed (changes requested):
Just a question about how the protocol is derived

LinHungWu commented:
What is the default value if there is nothing to derived to?

jeddcuison committed:
Use "http:" as the default protocol if the status path is relative

In addition:

- check for "https:" also when deriving the protocol from the status path
- update the corresponding Selenium test

jeddcuison commented:
Answer to @LinHungWu's question "What is the default value if there is nothing to derived to?"

As discussed with @akinny, the default is "http:"

akinny commented on jwala-webapp/src/main/webapp/resources/js/react/group-operations/WebServerControlPanelWidget.js:
```diff
@@ -108,7 +108,10 @@ var WebServerControlPanelWidget = React.createClass({
     onClickStatusLink: function(e) {
-        let statusUrl = "https://" + this.props.data.host + ":" + this.props.data.httpsPort + jwalaVars.loadBalancerStatusMount
+        let locationProtocol =  WebServerControlPanelWidget.parseUrlProtocol(this.props.data.statusPath.path);
+        let port = locationProtocol === WebServerControlPanelWidget.HTTP_PROTOCOL ? this.props.data.port :
```
shouldn't this.props.data.port be
this.props.data.httpPort?

akinny committed:
Merge branch 'master' into derive-location-protocol-from-status-path-for-balancer-manager
